src/main/java/mate/academy/hibernate/relations/dao/impl/CountryDaoImpl.java

```diff
@@ -12,11 +15,29 @@
 
     @Override
     public Country add(Country country) {
-        return null;
+        Session session = factory.openSession();
+        if (session == null) {
+            throw new DataProcessingException("Session not initialized");
+        }
+        Transaction transaction = session.beginTransaction();
```

Comment:
According to the checklist, you should use try with resources for read operations. Although this is a write operation, it's good practice to use try-with-resources to ensure that the session is always closed, even if an exception is thrown. This will make the code more robust and consistent.